python/cudf/cudf/core/column/numerical.py

```diff
@@ -161,17 +162,21 @@ def as_numerical_column(self, dtype, **kwargs):
         return libcudf.unary.cast(self, dtype)
 
     def to_pandas(self, index=None):
-        if self.has_nulls and self.dtype == np.bool:
-            # Boolean series in Pandas that contains None/NaN is of dtype
-            # `np.object`, which is not natively supported in GDF.
-            ret = self.astype(np.int8).fillna(-1).to_array()
-            ret = pd.Series(ret, index=index)
-            ret = ret.where(ret >= 0, other=None)
-            ret.replace(to_replace=1, value=True, inplace=True)
-            ret.replace(to_replace=0, value=False, inplace=True)
-            return ret
+
+        arrow_data = self.to_arrow()
+        host_dtype = pd_dtype_from_cudf_dtype(
+            self.dtype, has_nulls=self.has_nulls
+        )
+        if hasattr(host_dtype, "__from_arrow__"):
+            pandas_array = host_dtype.__from_arrow__(arrow_data)
+
         else:
-            return pd.Series(self.to_array(fillna="pandas"), index=index)
+            pandas_array = arrow_data.to_pandas()
+
+        result = pd.Series(pandas_array, copy=True)
```

Comment:
Why do we need to copy here?

Reply:
The reason is that if we don't pass `copy=True`, pandas ends up creating a read-only pandas series. Here is an example:

```
>>> arr = pa.array([1,2])
>>> sr = pd.Series(arr)
>>> sr[0] = 3
*snip*
ValueError: assignment destination is read-only
```

Reply:
Got it. Okay, instead of using `to_arrow` and going through Arrow for these nullable types, we should probably expand the mask from bits to bools on the GPU and then construct the pandas array from host buffers ourselves zero copy.

Reply:
That was my original approach as well, basically the reverse of the `from_pandas` process. But I thought the whole point was to go through arrow as much as possible?